### PR TITLE
Add result_type field to TimingStats and DatafeedTimingStats documents

### PR DESCRIPTION
**Pull request by przemekwitek**
Having result_type field persisted improves debuggability.

### CODE REVIEW

File: x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/ml/job/process/autodetect/state/TimingStats.java [hunk 1]
```diff
@@ -195,6 +196,9 @@ public void writeTo(StreamOutput out) throws IOException {
     @Override
     public XContentBuilder toXContent(XContentBuilder builder, Params params) throws IOException {
         builder.startObject();
+        if (params.paramAsBoolean(ToXContentParams.FOR_INTERNAL_STORAGE, true)) {
```
Comment (davidkyle) [inline x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/ml/job/process/autodetect/state/TimingStats.java]: Most other classes that test for this field e.g [DataFrameAnalyticsConfig](https://github.com/elastic/elasticsearch/blob/master/x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/ml/dataframe/DataFrameAnalyticsConfig.java#L205) default to value to false and only set the param when saving the document.

**Review by davidkyle — APPROVED** [review]
LGTM

I left a comment about the default value of the field and making it more consistent with other usages.

File: x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/ml/job/process/autodetect/state/TimingStats.java (hunk 1 above)
Comment (przemekwitek) [reply x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/ml/job/process/autodetect/state/TimingStats.java]: Done.

**Review by davidkyle — APPROVED** [review]
LGTM

**Pull request merged**